From 481458aabe001a111d553a5a6f46f38658ef22e0 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Thu, 1 Sep 2022 11:22:17 +0800
Subject: [PATCH 2/5] enable ecjpake for ffs

---
 include/mbedtls/config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/mbedtls/config.h b/include/mbedtls/config.h
index 1cd6eb6..6755eb2 100644
--- a/include/mbedtls/config.h
+++ b/include/mbedtls/config.h
@@ -2814,7 +2814,7 @@
  *
  * Requires: MBEDTLS_ECP_C, MBEDTLS_MD_C
  */
-//#define MBEDTLS_ECJPAKE_C
+#define MBEDTLS_ECJPAKE_C
 
 /**
  * \def MBEDTLS_ECP_C
-- 
2.20.1

