From 4fae08e6d487b0614af98d0f6517c16c47ff51f2 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 4 Mar 2019 17:22:51 +0800
Subject: [PATCH 14/23] qwaylandwindow: Don't try to move fullscreen/maximized
 window

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 src/client/qwaylandwindow.cpp | 5 ++++-
 1 file changed, 4 insertions(+), 1 deletion(-)

diff --git a/src/client/qwaylandwindow.cpp b/src/client/qwaylandwindow.cpp
index 0a02866d..dd667c6d 100644
--- a/src/client/qwaylandwindow.cpp
+++ b/src/client/qwaylandwindow.cpp
@@ -329,7 +329,10 @@ void QWaylandWindow::setGeometry_helper(const QRect &rect)
 
 void QWaylandWindow::setGeometry(const QRect &rect)
 {
-    setGeometry_helper(rect);
+    if (mState == Qt::WindowFullScreen || mState == Qt::WindowMaximized)
+        setGeometry_helper(QRect(QPoint(), rect.size()));
+    else
+        setGeometry_helper(rect);
 
     if (window()->isVisible() && rect.isValid()) {
         if (mWindowDecoration)
-- 
2.20.1

