From 288e1f54c21f172ac167d750fbfad2b309d8e73f Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Tue, 22 Feb 2022 09:51:29 +0800
Subject: [PATCH 67/94] HACK: compositor: Re-build view list when committing an
 unassigned surface

To avoid skipping repaint for unassigned surface.

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/compositor.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/libweston/compositor.c b/libweston/compositor.c
index f6781caaba..41701189db 100644
--- a/libweston/compositor.c
+++ b/libweston/compositor.c
@@ -4673,6 +4673,10 @@ weston_surface_commit(struct weston_surface *surface)
 	if (status & WESTON_SURFACE_DIRTY_SUBSURFACE_CONFIG)
 		weston_surface_commit_subsurface_order(surface);
 
+	/* HACK: Assign outputs */
+	if (!surface->output_mask)
+		weston_compositor_build_view_list(surface->compositor);
+
 	weston_surface_schedule_repaint(surface);
 
 	return status;
-- 
2.20.1

