From feef7902a390a7e83fb43f1134acec06fcd3190b Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 28 Sep 2022 10:44:07 +0800
Subject: [PATCH 081/100] compositor: Stop bootanim when seen first app

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 libweston/compositor.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/libweston/compositor.c b/libweston/compositor.c
index bfa7286..911d13d 100644
--- a/libweston/compositor.c
+++ b/libweston/compositor.c
@@ -3103,6 +3103,10 @@ weston_compositor_build_view_list(struct weston_compositor *compositor,
 						view->surface->flags |=
 							SURFACE_STAY_ON_BOTTOM;
 				}
+
+#define BOOTANIM "/usr/bin/bootanim"
+				if (!access(BOOTANIM, X_OK))
+					(void)!system(BOOTANIM " stop&");
 			}
 
 			view_list_add(compositor, view, output);
@@ -3386,6 +3390,7 @@ output_repaint_timer_handler(void *data)
 
 	if (!access(getenv("WESTON_FREEZE_DISPLAY") ? : "", F_OK)) {
 		usleep(DEFAULT_REPAINT_WINDOW * 1000);
+		weston_compositor_build_view_list(compositor, NULL);
 		goto out;
 	}
 
-- 
2.20.1

