From 2d531970d4e05032ebbb186923fef114feb7e1bd Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Sun, 13 Aug 2023 11:25:25 +0800
Subject: [PATCH 84/98] desktop-shell: Always respawn when disconnected

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 desktop-shell/shell.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/desktop-shell/shell.c b/desktop-shell/shell.c
index 402fe57b..b5e3af47 100644
--- a/desktop-shell/shell.c
+++ b/desktop-shell/shell.c
@@ -4258,11 +4258,13 @@ respawn_desktop_shell_process(struct desktop_shell *shell)
 		shell->child.deathcount = 0;
 	}
 
+#if 0
 	shell->child.deathcount++;
 	if (shell->child.deathcount > 5) {
 		weston_log("%s disconnected, giving up.\n", shell->client);
 		return;
 	}
+#endif
 
 	weston_log("%s disconnected, respawning...\n", shell->client);
 	launch_desktop_shell_process(shell);
-- 
2.20.1

