From 78a73dea1b641d16fc1598b90c2546481ae501b0 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Sat, 7 Oct 2023 17:08:03 +0800
Subject: [PATCH 2/6] adb: daemon: Fix build issue with uclibc

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 adb/adbd_auth.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/adb/adbd_auth.cpp b/adb/adbd_auth.cpp
index 3488ad1..c78c9cb 100644
--- a/adb/adbd_auth.cpp
+++ b/adb/adbd_auth.cpp
@@ -67,7 +67,7 @@ bool adbd_auth_verify(const char* token, size_t token_size, const char* sig, int
                 // b64_pton requires one additional byte in the target buffer for
                 // decoding to succeed. See http://b/28035006 for details.
                 uint8_t keybuf[ANDROID_PUBKEY_ENCODED_SIZE + 1];
-                if (__b64_pton(line.c_str(), keybuf, sizeof(keybuf)) != ANDROID_PUBKEY_ENCODED_SIZE) {
+                if (b64_pton(line.c_str(), keybuf, sizeof(keybuf)) != ANDROID_PUBKEY_ENCODED_SIZE) {
                     LOG(ERROR) << "Invalid base64 key " << line.c_str() << " in " << path;
                     continue;
                 }
-- 
2.20.1

