From 86faa64dd525355920a726a62be60e49fa9abb70 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Mon, 1 Apr 2024 10:27:25 +0800
Subject: [PATCH 2/4] egl.h: Remove unused eglQueryWaylandBufferWL

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 include/render/egl.h | 1 -
 1 file changed, 1 deletion(-)

diff --git a/include/render/egl.h b/include/render/egl.h
index 36b70ae..b690877 100644
--- a/include/render/egl.h
+++ b/include/render/egl.h
@@ -31,7 +31,6 @@ struct wlr_egl {
 		PFNEGLGETPLATFORMDISPLAYEXTPROC eglGetPlatformDisplayEXT;
 		PFNEGLCREATEIMAGEKHRPROC eglCreateImageKHR;
 		PFNEGLDESTROYIMAGEKHRPROC eglDestroyImageKHR;
-		PFNEGLQUERYWAYLANDBUFFERWL eglQueryWaylandBufferWL;
 		PFNEGLQUERYDMABUFFORMATSEXTPROC eglQueryDmaBufFormatsEXT;
 		PFNEGLQUERYDMABUFMODIFIERSEXTPROC eglQueryDmaBufModifiersEXT;
 		PFNEGLDEBUGMESSAGECONTROLKHRPROC eglDebugMessageControlKHR;
-- 
2.20.1

