From 90bc7e00c221853ebc23ab794f4fba11f0cf016c Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 1 Apr 2020 09:17:45 +0800
Subject: [PATCH 12/14] Enable v4l2 stuff by default

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 chromium/media/gpu/args.gni | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/chromium/media/gpu/args.gni b/chromium/media/gpu/args.gni
index 7e905aebb..14bf4a3d4 100644
--- a/chromium/media/gpu/args.gni
+++ b/chromium/media/gpu/args.gni
@@ -4,15 +4,15 @@
 
 declare_args() {
   # Indicates if V4L plugin is used.
-  use_v4lplugin = false
+  use_v4lplugin = true
 
   # Indicates if Video4Linux2 codec is used. This is used for all CrOS
   # platforms which have v4l2 hardware encoder / decoder.
-  use_v4l2_codec = false
+  use_v4l2_codec = true
 
   # Indicates that only definitions available in the mainline linux kernel
   # will be used.
-  use_linux_v4l2_only = false
+  use_linux_v4l2_only = true
 
   # Indicates if Video4Linux2 AML encoder is used. This is used for AML
   # platforms which have v4l2 hardware encoder
-- 
2.20.1

