From 9e2eb604fa9846df750facdc4d13177b8972e751 Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Wed, 25 Nov 2020 10:51:12 +0800
Subject: [PATCH 60/78] waylandsink: Drop frame when window not ready

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 ext/wayland/gstwaylandsink.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/ext/wayland/gstwaylandsink.c b/ext/wayland/gstwaylandsink.c
index d5e1526..b75687b 100644
--- a/ext/wayland/gstwaylandsink.c
+++ b/ext/wayland/gstwaylandsink.c
@@ -869,10 +869,15 @@ render:
 
 no_window_size:
   {
+    /* HACK: Drop frame when window not ready */
+#if 0
     GST_ELEMENT_ERROR (sink, RESOURCE, WRITE,
         ("Window has no size set"),
         ("Make sure you set the size after calling set_window_handle"));
     ret = GST_FLOW_ERROR;
+#else
+    GST_WARNING_OBJECT (sink, "Window has no size set");
+#endif
     goto done;
   }
 no_buffer:
-- 
2.20.1

