From 870117aa9fa421ed11979aada6dada63c914bc63 Mon Sep 17 00:00:00 2001
From: Yao Xiao <xiaoyao@rock-chips.com>
Date: Tue, 21 Nov 2023 14:40:09 +0800
Subject: [PATCH 3/3] fixed avdtp Transaction Label Number

---
 profiles/audio/avdtp.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/profiles/audio/avdtp.c b/profiles/audio/avdtp.c
index 3667e08..392ceea 100644
--- a/profiles/audio/avdtp.c
+++ b/profiles/audio/avdtp.c
@@ -286,6 +286,7 @@ struct in_buf {
 	gboolean active;
 	int no_of_packets;
 	uint8_t transaction;
+	uint8_t received_cmd_transaction;
 	uint8_t signal_id;
 	uint8_t buf[1024];
 	uint8_t data_size;
@@ -1469,7 +1470,7 @@ static void setconf_cb(struct avdtp *session, struct avdtp_stream *stream,
 		return;
 	}
 
-	if (!avdtp_send(session, session->in_cmd.transaction,
+	if (!avdtp_send(session, session->in_cmd.received_cmd_transaction,
 			AVDTP_MSG_TYPE_ACCEPT,
 			AVDTP_SET_CONFIGURATION, NULL, 0)) {
 		stream_free(stream);
@@ -2252,6 +2253,7 @@ static gboolean session_cb(GIOChannel *chan, GIOCondition cond,
 	}
 
 	if (header->message_type == AVDTP_MSG_TYPE_COMMAND) {
+		session->in_cmd.received_cmd_transaction = session->in_cmd.transaction;
 		if (!avdtp_parse_cmd(session, session->in_cmd.transaction,
 				     session->in_cmd.signal_id,
 				     session->in_cmd.buf,
-- 
2.20.1

