From c6e102ab79b1b19451ad9e92657e51d47cc5ec8a Mon Sep 17 00:00:00 2001
From: Jeffy Chen <jeffy.chen@rock-chips.com>
Date: Fri, 10 Jun 2022 17:15:12 +0800
Subject: [PATCH 07/11] Skip drawing when cwidth is zero

Signed-off-by: Jeffy Chen <jeffy.chen@rock-chips.com>
---
 term.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/term.c b/term.c
index c068d03..8e5a747 100644
--- a/term.c
+++ b/term.c
@@ -98,6 +98,9 @@ static int term_draw_cell(struct tsm_screen* screen, uint64_t id,
 	if (age && terminal->term->age && age <= terminal->term->age)
 		return 0;
 
+	if (!cwidth)
+		return 0;
+
 	if (terminal->background_valid) {
 		br = (terminal->background >> 16) & 0xFF;
 		bg = (terminal->background >> 8) & 0xFF;
-- 
2.20.1

