From fb347cd9d4c543e4ca409fc2324f57161567fae1 Mon Sep 17 00:00:00 2001
From: "jkand.huang" <jkand.huang@rock-chips.com>
Date: Mon, 23 Mar 2020 17:02:38 +0800
Subject: [PATCH 2/7] live555: If no session is found, the first session is
 returned.

Signed-off-by: jkand.huang <jkand.huang@rock-chips.com>
---
 liveMedia/GenericMediaServer.cpp | 9 ++++++++-
 1 file changed, 8 insertions(+), 1 deletion(-)

diff --git a/liveMedia/GenericMediaServer.cpp b/liveMedia/GenericMediaServer.cpp
index f4aaa8d..da8a66a 100644
--- a/liveMedia/GenericMediaServer.cpp
+++ b/liveMedia/GenericMediaServer.cpp
@@ -391,7 +391,14 @@ GenericMediaServer::lookupClientSession(char const* sessionIdStr) {
 }
 
 ServerMediaSession* GenericMediaServer::getServerMediaSession(char const* streamName) {
-  return (ServerMediaSession*)(fServerMediaSessions->Lookup(streamName));
+  // if not found, return the first.
+  ServerMediaSession *session =
+      (ServerMediaSession *)(fServerMediaSessions->Lookup(streamName));
+  if (session == NULL) {
+    session = (ServerMediaSession *)(fServerMediaSessions->getFirst());
+  }
+  return session;
+//  return (ServerMediaSession*)(fServerMediaSessions->Lookup(streamName));
 }
 
 
-- 
2.20.1

