From 35a06eb08bd61e68e8e1c783f41d0c9d0869dbbd Mon Sep 17 00:00:00 2001
From: "jkand.huang" <jkand.huang@rock-chips.com>
Date: Wed, 17 Jun 2020 17:05:31 +0800
Subject: [PATCH 7/7] live555: fix close fd error.

Signed-off-by: jkand.huang <jkand.huang@rock-chips.com>
---
 liveMedia/RTSPServer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/liveMedia/RTSPServer.cpp b/liveMedia/RTSPServer.cpp
index 5932225..c2f8170 100644
--- a/liveMedia/RTSPServer.cpp
+++ b/liveMedia/RTSPServer.cpp
@@ -609,7 +609,7 @@ void RTSPServer::RTSPClientConnection::closeSocketsRTSP() {
   fOurRTSPServer.stopTCPStreamingOnSocket(fClientOutputSocket);
 
   // Turn off background handling on our input socket (and output socket, if different); then close it (or them):
-  if (fClientOutputSocket != fClientInputSocket) {
+  if (fClientOutputSocket != fClientInputSocket && fClientInputSocket != -1) {
     envir().taskScheduler().disableBackgroundHandling(fClientOutputSocket);
     ::closeSocket(fClientOutputSocket);
   }
-- 
2.20.1

